app/src/test/java/com/android/sample/model/workout/BodyWeightWorkoutTest.kt

```diff
@@ -0,0 +1,37 @@
+package com.android.sample.model.workout
+
+import org.junit.Assert.assertEquals
+import org.junit.Test
+
+class BodyWeightWorkoutTest {
+
+  @Test
+  fun testBodyWeightWorkoutCreation() {
+    val workout =
+        BodyWeightWorkout(
+            name = "Bodyweight Strength",
+            description = "A workout focused on bodyweight exercises.",
+            warmup = true)
+
+    assertEquals("Bodyweight Strength", workout.name)
+    assertEquals("A workout focused on bodyweight exercises.", workout.description)
+    assertEquals(true, workout.warmup)
+    assertEquals(0, workout.exercises.size) // Initially, exercises should be empty
+  }
+
+  @Test
+  fun testAddExercise() {
+    val workout =
+        BodyWeightWorkout(
+            name = "Bodyweight Strength",
+            description = "A workout focused on bodyweight exercises.",
+            warmup = true)
+
+    val pushUps =
+        BodyWeightExercise(BodyWeightExerciseType.PUSH_UPS, ExerciseDetail.RepetitionBased(15))
+    workout.addExercise(pushUps)
+
+    assertEquals(1, workout.exercises.size) // Now should have 1 exercise
+    assertEquals(pushUps, workout.exercises[0]) // Check if the exercise is correctly added
+  }
+}
```

Comment:
The test looks good but it might also be beneficial to check cases where an exercise is added to a list that's already initialized with some exercises. This would ensure the addExercise method behaves as expected with a pre-populated list.